File: 0485_max_consecutive_ones/solution.ts
// LeetCode 0485 - Max Consecutive Ones
// https://leetcode.com/problems/max-consecutive-ones/

export class Solution {
    findMaxConsecutiveOnes(nums: number[]): number {
        let best = 0;
        let current = 0;
        for (const num of nums) {
            if (num === 1) {
                current += 1;
                best = Math.max(best, current);
            } else {
                current = 0;
            }
        }
        return best;
    }
}

File: 0485_max_consecutive_ones/solution.go
// LeetCode 0485 - Max Consecutive Ones
// https://leetcode.com/problems/max-consecutive-ones/

func findMaxConsecutiveOnes(nums []int) int {
	best := 0
	current := 0
	for _, num := range nums {
		if num == 1 {
			current++
			if current > best {
				best = current
			}
		} else {
			current = 0
		}
	}
	return best
}
